magicctapipe/image/calib.py

```diff
@@ -0,0 +1,250 @@
+    calibrator_magic: CameraCalibrator (ctapipe.calib)
+        ctapipe object needed to calibrate the camera
+    
+
+    Returns
+    -------
+    signal_pixels: Mask of the pixels selected by the cleaning
+    image: Array of number of p.e. in the camera pixels
+    peak_time: Array of the signal peak time in the camera pixels
+    """
+    
+    calibrator_magic._calibrate_dl0(event, tel_id)
+    calibrator_magic._calibrate_dl1(event, tel_id)
+
+    image = event.dl1.tel[tel_id].image.astype(np.float64)
+    peak_time = event.dl1.tel[tel_id].peak_time.astype(np.float64)
+    use_charge_correction = config_magic["charge_correction"]["use"]
+    
+    if use_charge_correction:
+        # Scale the charges by the correction factor
+        image *= config_magic["charge_correction"]["factor"]
+
+    # Apply the image cleaning
+    signal_pixels, image, peak_time = magic_clean[tel_id].clean_image(
+        event_image=image, event_pulse_time=peak_time
+    )
+    return signal_pixels, image, peak_time
+
+
+
+
+
+def Calibrate(event, tel_id, config, calibrator, LST_bool, obs_id=None, camera_geoms=None, magic_clean=None):
+
+    """
+    This function computes and returns some information for a single event of a telescope 
+
+    Parameters
+    ----------
+    event: event 
+        From an EventSource
+    tel_id: int
+        Telescope ID     
+    config: dictionary
+        Parameters for image extraction and calibration    
+    calibrator: CameraCalibrator (ctapipe.calib)
+        ctapipe object needed to calibrate the camera
+    LST_bool: bool
+        Whether the telescope is a LST
+    obs_id: int
+        Observation ID. Unsed in case of LSt telescope
+    camera_geoms: telescope.camera.geometry
+        Camera geometry. Used in case of LST telescope
+    magic_clean: dictionary (1 entry per MAGIC telescope)
+        Each entry is a MAGICClean object using the telescope camera geometry. Used in case of MAGIC telescope
+    
+
+    Returns
+    -------
+    signal_pixels: Mask of the pixels selected by the cleaning
+    image: Array of number of p.e. in the camera pixels
+    peak_time: Array of the signal peak time in the camera pixels
+
+    """
+    
+    calibrator._calibrate_dl0(event, tel_id)
+    calibrator._calibrate_dl1(event, tel_id)
+
+    image = event.dl1.tel[tel_id].image.astype(np.float64)
+    peak_time = event.dl1.tel[tel_id].peak_time.astype(np.float64)
+    if LST_bool==False: 
+        use_charge_correction = config["charge_correction"]["use"]
+    
+        if use_charge_correction:
+            # Scale the charges by the correction factor
+            image *= config["charge_correction"]["factor"]
+        # Apply the image cleaning
+        signal_pixels, image, peak_time = magic_clean[tel_id].clean_image(
+            event_image=image, event_pulse_time=peak_time
+        )
+    else:
+        increase_nsb = config["increase_nsb"].pop("use")
```

Comment:
I would add here an if on obs_id, camera_geoms or magic_clean being null to throw an exception (if somebody tries to run the function on LST without providing needed information)

Reply:
Done (?)